core/src/main/java/org/elasticsearch/common/unit/DistanceUnit.java

```diff
@@ -324,23 +296,30 @@ private static Distance parseDistance(String distance, DistanceUnit defaultUnit)
         }
     }
 
-    private static final DistanceUnit PROTOTYPE = DEFAULT;
+    /**
+     * Read a {@link DistanceUnit} from a {@link StreamInput}.
+     *
+     * @param in {@link StreamInput} to read the {@link DistanceUnit} from
+     * @return {@link DistanceUnit} read from the {@link StreamInput}
+     * @throws IOException if no unit can be read from the {@link StreamInput}
+     * @throws IllegalArgumentException if no matching {@link DistanceUnit} can be found
+     */
+    public static DistanceUnit readFromStream(StreamInput in) throws IOException {
+        byte b = in.readByte();
 
-    @Override
-    public DistanceUnit readFrom(StreamInput in) throws IOException {
-        int ordinal = in.readVInt();
-        if (ordinal < 0 || ordinal >= values().length) {
-            throw new IOException("Unknown DistanceUnit ordinal [" + ordinal + "]");
+        if (b < 0 || b >= values().length) {
+            throw new IllegalArgumentException("No type for distance unit matching [" + b + "]");
         }
-        return values()[ordinal];
-    }
-
-    public static DistanceUnit readUnitFrom(StreamInput in) throws IOException {
-        return PROTOTYPE.readFrom(in);
+        return values()[b];
     }
 
+    /**
+     * Write a {@link DistanceUnit} to a {@link StreamOutput}.
+     *
+     * @param out {@link StreamOutput} to write to
+     */
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        out.writeVInt(this.ordinal());
+        out.writeByte((byte) this.ordinal());
```

Comment:
Yeah!  You are right!  Maybe I should just let it stay vint then. I'm
pretty sure it is impossible to have more enum constants than can fit in a
short, btw. I looked it up!
On Mar 25, 2016 5:23 PM, "Luca Cavanna" <notifications@github.com> wrote:

> In core/src/main/java/org/elasticsearch/common/unit/DistanceUnit.java
> <https://github.com/elastic/elasticsearch/pull/17336#discussion_r57492779>
> :
>
> >      @Override
> >      public void writeTo(StreamOutput out) throws IOException {
> > -        out.writeVInt(this.ordinal());
> > +        out.writeByte((byte) this.ordinal());
>
> sure I was more looking for best practices around this. cause VInt doesn't
> waste space anyways right so there isn't a huge difference practically?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/17336/files/b77b15dcf9fee4669ac5b30b5e795fb414d9a83c#r57492779>
>